plugins/out_es/es.c

```diff
@@ -886,12 +886,24 @@ static void cb_es_flush(struct flb_event_chunk *event_chunk,
                 if (ctx->trace_error) {
                     /*
                      * If trace_error is set, trace the actual
-                     * input/output to Elasticsearch that caused the problem.
+                     * response from Elasticsearch explaining the problem.
+                     * Trace_Output can be used to see the request. 
                      */
-                    flb_plg_debug(ctx->ins, "error caused by: Input\n%s\n",
-                                  pack);
-                    flb_plg_error(ctx->ins, "error: Output\n%s",
-                                  c->resp.payload);
+                    if (pack_size < 4000) {
```

Comment:
Shouldn't this be `4096` or are you purposefully short of exactly 4KB?

Reply:
I need to test this if we go with it. But yea that was the thought- I think it needs to be less than the actual limit, but need to check it by testing. 

The current recommended solution from Takahiro & Eduardo here is to iteratively write 4KB blocks of the response via the flb_log function... by creating new functions that return the actual number of bytes processed, which I bet is less than 4KB since each message gets the timestamp and plugin name added to it...